python/heterocl/tvm/runtime.py

```diff
@@ -250,22 +258,48 @@ def copy_and_compile(platform, mode, backend, host_only, cfg, tcl):
 
     elif platform == "aocl":
         env = os.environ.copy()
+
+        # check aoc version 
+        assert os.system("which aoc >> /dev/null") == 0, \
+            "cannot find aoc on system path"
+        ver = run_process("aoc --version", "\d+\.\d\.\d")[0].split(".")
+
         assert "INTELFPGAOCLSDKROOT" in os.environ, \
                "cannot find aocl sdk for fpga on path" 
 
         os.system("cp " + path + "aocl/* project/")
         cmd = "cd project; make clean; make;"
+
         # compile kernel for xcel device
         cmd += " aoc"
         if mode == "sw_sim":
             cmd += " -march=emulator"
+        elif mode == "hw_sim":
+            if int(ver[0]) < 19:
+                raise RuntimeError("AOC version {}.{}.{} too old, ".format(*ver) + \
```

Comment:
Check the grammar of the error messages. AOC version ... too old => AOC version ... is too old.